other/autoregressive-process/utils.jl

```diff
@@ -0,0 +1,42 @@
+# Generates an AR(p) process with coefficients `ϕ`.
+# `ϕ` should be provided as a vector and it represents the coefficients of the AR model.
+# Hence the order of the generated process is equal to the length of `ϕ`.
+# `s` indicates the total length of the series to be generated.
+function generate_process(ϕ::AbstractVector{Float32}, s::Int)
+    @assert s > 0 "s must be positive"
+    # Generate white noise
+    ϵ = randn(Float32, s)
+    # Initialize time series
+    X = zeros(Float32, s)
+    p = length(ϕ)
+    X[1] = ϵ[1]
+    # Reverse the order of the coefficients for multiplication later on
+    reverse!(ϕ) 
+    # Compute values iteratively
+    for t ∈ 2:s
+        if t ≤ p
+            X[t] = sum(X[1:t-1] .* ϕ[1:t-1]) + ϵ[t]
+        else
+            X[t] = sum(X[t-p:t-1] .* ϕ) + ϵ[t]
+        end
+    end
```

Comment:
This can definitely be simplified, I'm not exactly sure of how much though. Can the loop be expressed as a convolution?

Reply:
I made the following change
```julia
# Fill first p observations
for t ∈ 1:p-1
    X[t+1] = X[1:t]'ϕ[1:t] + ϵ[t+1]
end
# Compute values iteratively
for t ∈ p+1:s
    X[t] = X[t-p:t-1]'ϕ + ϵ[t]
end
```
which is a bit faster. I don't think I can use a `mapreduce` here but perhaps someone more knowledgeable can enlighten me.